### ISSUE TITLE
Build Failure Due to Duplicate libreactnative.so File in Realm and React-Android

### ISSUE THREAD

**Issue by Jean-Diniz** [realm/realm-js#6918]
### How frequently does the bug occur?

Always

### Description

When building the app with Gradle, the build process fails due to a duplicate libreactnative.so file. I expected the app to build and install without issues, but instead, the build fails with an error indicating that there are two instances of libreactnative.so, one from the realm library and another from react-android. This conflict prevents the app from installing.

### Stacktrace & log output

```shell
Execution failed for task ':app:mergeDebugNativeLibs'.
> A failure occurred while executing com.android.build.gradle.internal.tasks.MergeNativeLibsTask$MergeNativeLibsTaskWorkAction 
> 2 files found with path 'lib/arm64-v8a/libreactnative.so' from inputs: 
-/home/foobar/project/node_modules/realm/binding/android/build/intermediates/library_jni/debug/copyDebugJniLibsProjectOnly/jni/arm64-v8a/libreactnative.so 
- /home/foobar/.gradle/caches/8.10.2/transforms/095f498f1545b2a557a515b33db49b37/transformed/jetified-react-android-0.76.1-debug/jni/arm64-v8a/libreactnative.so
```

### Can you reproduce the bug?

Always

### Reproduction Steps

```json
{
  "name": "foobar",
  "version": "0.0.0",
  "private": true,
  "packageManager": "yarn@3.6.4",
  "engines": {
    "node": "18.17.0"
  },
  "scripts": {
    "android": "APP_ENV=development react-native run-android",
  },
  "dependencies": {
    "react": "18.3.1",
    "react-native": "0.76.1",
    "realm": "12.13.2",
  },
  "devDependencies": {
    "@babel/core": "7.25.2",
    "@babel/preset-env": "7.25.3",
    "@babel/preset-typescript": "7.18.6",
    "@babel/runtime": "7.25.0",
    "@react-native-community/cli": "15.0.0",
    "@react-native-community/cli-platform-android": "15.0.0",
    "@react-native-community/cli-platform-ios": "15.0.0",
    "@react-native/babel-preset": "0.76.1",
    "@react-native/eslint-config": "0.76.1",
    "@react-native/metro-config": "0.76.1",
    "@react-native/typescript-config": "0.76.1",
  }
}
```
 - cd android
 - ./gradlew clean && cd ../
 - yarn
 - yarn android

### Version

12.13.2

### What services are you using?

Local Database only

### Are you using encryption?

No

### Platform OS and version(s)

Android 12

### Build environment

OS: [Ubuntu 22.04]
Node.js: 18.17.0
React Native: 0.76.1
Gradle: 8.10.2
Java 17.0.7-zulu

**Comment by mtalhadev** [comment]
For temporary fixing it, you can do this:

./node_modules/realm/binding/android/build.gradle

Add this one line in the excludes array.
    packagingOptions {
        excludes = [
                "META-INF",
                "META-INF/**",
                "**/libc++_shared.so",
                "**/libfbjni.so",
                "**/libjsi.so",
                "**/libreactnativejni.so",
                "**/libturbomodulejsijni.so",
                "**/libreactnative.so", //Add this line
        ]
    }

**Comment by joergjaeckel** [comment]
For whatever reason I also had to change `"/libjsi.so"` to `"**/libjsi.so"`.

**Comment by edritech93** [comment]
still issue
"react-native": "0.76.1",
 "realm": "12.13.2",

**Comment by kraenhansen** [comment]
@edritech93 right. The PR is merged, however this is yet to be released.

**Comment by kraenhansen** [comment]
The fix for this was picked onto the `community` branch and released as [v20.1.0](https://github.com/realm/realm-js/releases/tag/v20.1.0)
